pxe/common_files/disk_imaging.sh

```
@@ -314,11 +339,135 @@ function modify_grub_cfg() {
+	esp_part_num=$(get_part_num "$efi_dev")
+	if [ -z "$esp_part_num" ]; then
+		echo "Could not determine ESP partition number for $efi_dev, skipping EFI boot entry creation" | tee $log_output
+		return 0
+	fi
+
+	if efibootmgr | grep -qi "^Boot[0-9A-Fa-f]\{4\}\*\?[[:space:]]*$label\$"; then
+		echo "EFI boot entry for $label already exists, skipping creation" | tee $log_output
+		return 0
+	fi
+
+	echo "Creating EFI boot entry for $label ($loader_path on $image_disk part $esp_part_num)" | tee $log_output
```

Shellcheck (0.11.0)

[info] Double quote to prevent globbing and word splitting.

(SC2086)